plugin/trino-hive/src/test/java/io/trino/plugin/hive/parquet/AbstractTestParquetReader.java

```diff
@@ -1988,6 +1995,85 @@ public void testMapMaxReadBytes()
+                    .result().matches(resultBuilder(queryRunner.getDefaultSession(), new ArrayType(VARCHAR))
+                            .row(Arrays.asList("hello", "world"))
+                            .row(Arrays.asList("good", "bye"))
+                            .row(Arrays.asList("one", "two", "three"))
+                            .build());
+            queryRunner.execute("DROP TABLE table_with_repeated_primitive_string");
+        }
+    }
+
+    @Test
+    public void testBackwardsCompatibleRepeatedIntegerField()
+            throws Exception
+    {
+        File parquetFile = new File(Resources.getResource("parquet_repeated_primitives/int/").toURI());
+        try (DistributedQueryRunner queryRunner = HiveQueryRunner.builder().build();
+                QueryAssertions assertions = new QueryAssertions(queryRunner)) {
+            queryRunner.execute(format("""
+                        CREATE TABLE table_with_repeated_primitive_int (
+                            repeatedInt array<int>)
+                        WITH (
+                            external_location = '%s',
+                            format = 'PARQUET')
+                            """,
+                    parquetFile.getAbsolutePath()));
+
+            assertThat(assertions.query("SELECT repeatedInt FROM table_with_repeated_primitive_int"))
+                    .result().matches(resultBuilder(queryRunner.getDefaultSession(), new ArrayType(INTEGER))
+                            .row(Arrays.asList(1, 2, 3))
+                            .build());
+            queryRunner.execute("DROP TABLE table_with_repeated_primitive_int");
+        }
+    }
+
+    @Test
+    public void testBackwardsCompatibleRepeatedPrimitiveFieldDefinedAsPrimitive()
+            throws Exception
+    {
+        DistributedQueryRunner queryRunner = HiveQueryRunner.builder().build();
+        File parquetFile = new File(Resources.getResource("parquet_repeated_primitives/int/").toURI());
+        boolean errorThrown = false;
+        try {
+            queryRunner.execute(format("""
+                        CREATE TABLE table_with_repeated_primitive_int_bad_table_schema (
+                            repeatedInt int)
+                        WITH (
+                            external_location = '%s',
+                            format = 'PARQUET')
+                            """,
+                    parquetFile.getAbsolutePath()));
+            queryRunner.execute("SELECT repeatedInt FROM table_with_repeated_primitive_int_bad_table_schema");
+        }
+        catch (QueryFailedException e) {
+            if (e.getMessage().contains("Repeated field repeatedint's type INT32 in a parquet file is incompatible with type integer defined in table schema")) {
+                errorThrown = true;
+            }
+        }
+        finally {
+            queryRunner.execute("DROP TABLE IF EXISTS table_with_repeated_primitive_int_bad_table_schema");
+        }
+        assert errorThrown;
```

Comment:
use `assertThatThrownBy(() -> ...)` instead, similar to what you see in `AbstractTestParquetReader#testParquetShortDecimalWriteToTrinoBigintBlockWithNonZeroScale` that includes assertions on the message and exception type.